### PR TITLE
Remove eth-keyring-controller from DecryptMessageController

### LINKED ISSUE #20724: Remove `@metamask/eth-keyring-controller` from `DecryptMessageController`

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

This PR removes the `keyringController` constructor options property, as the `KeyringController`'s method can now be invoked through the messaging system.

* Fixes #20724 
* See #20510 

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by cryptodev-2s — DISMISSED** [review]
LGTM!

File: app/scripts/controllers/decrypt-message.ts [hunk 1]
```diff
@@ -35,6 +36,27 @@ const stateMetadata = {
   unapprovedDecryptMsgCount: { persist: false, anonymous: false },
 };
 
+export const isEIP1024EncryptedMessage = (message: {
```
Comment (Gudahtt) [inline app/scripts/controllers/decrypt-message.ts:39]: Nit: Adding a JSDoc comment here would be great! Our normal lint rules require it for all declared functions

File: app/scripts/controllers/decrypt-message.ts [hunk 2]
```diff
@@ -35,6 +36,27 @@ const stateMetadata = {
...
+  from: string;
+  data: unknown;
+}): message is {
+  from: string;
+  data: Eip1024EncryptedData;
+} => {
+  if (
+    message.data &&
+    typeof message.data === 'object' &&
```
Comment (Gudahtt) [inline app/scripts/controllers/decrypt-message.ts:48]: Nit: Might be nice to use the `isObject` utility function here instead. What you have here is equivalent, but `isObject` is a bit easier to understand (readers might not understand that the truthy check is meant to rule out `null` which is also `typeof object`).
```suggestion
    isObject(message.data) && 
```

File: app/scripts/controllers/decrypt-message.ts [hunk 3]
```diff
@@ -216,7 +236,12 @@ export default class DecryptMessageController extends BaseControllerV2<
         await this._decryptMessageManager.approveMessage(messageParams);
 
       cleanMessageParams.data = this._parseMessageData(cleanMessageParams.data);
-      const rawMessage = await this._keyringController.decryptMessage(
+      if (!isEIP1024EncryptedMessage(cleanMessageParams)) {
+        throw new Error('Invalid encrypted data.');
```
Comment (Gudahtt) [inline app/scripts/controllers/decrypt-message.ts:240]: Could you add a test case for each of these new validation errors? It looks like they aren't yet captured

**Review by cryptodev-2s — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**